src/pages/index.js

```diff
@@ -70,31 +141,60 @@ function handleImageClick(name, link) {
   newImagePopup.open({ link, name });
 }
 
-const cardSection = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.getView();
   cardListEl.prepend(cardElement);
+  console.log(cardData);
+}
+
+function handleDeleteClick(card) {
+  newPopupConfirm.open();
+  newPopupConfirm.setSubmitAction(() => {
+    api.deleteCard(card.id).then(() => {
+      card.deleteCard();
+      newPopupConfirm.close();
+    });
+  });
 }
 
-cardSection.renderItems();
+function handleLikeClick(card) {
+  if (!card.isLiked) {
+    api.likeCard(card.id).then(() => {
+      card.updateLikes(true);
+    });
+  } else {
+    api.dislikeCard(card.id).then(() => {
+      card.updateLikes(false);
+    });
+  }
+}
 
 function handleProfileFormSubmit({ name, description }) {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
This is really cool, I am gonna leave this unresolved as I might try to refactor it to be that way later. Thank you for showing me =)